### PR TITLE
Support multiple components in same page

### PR DESCRIPTION
**Pull request by patw0929**
Related issue: #50 

The most code is copied from @xiankai's fork.
I had updated the example and tests.

### CODE REVIEW

File: __tests__/TelInput-test.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ describe('TelInput', () => {
 
   beforeEach(() => {
     component = TestUtils.renderIntoDocument(
-        <IntlTelInput css={['intl-tel-input', 'form-control phoneNumber']}
+        <IntlTelInput id={'0'} css={['intl-tel-input', 'form-control phoneNumber']}
```
Comment (brunops) [inline __tests__/TelInput-test.js]: Can decrease the burden on the user by using a uuid internally?

File: src/containers/App.js [hunk 2]
```diff
@@ -34,7 +35,8 @@ class IntlTelInput extends Component {
   render() {
     return (
       <Provider store={store}>
-        <IntlTelInputApp value={this.props.value}
+        <IntlTelInputApp id={this.props.id}
```
Comment (brunops) [inline src/containers/App.js]: Generate a `uuid` here, then the prop is not necessary .. see https://github.com/shtylman/node-uuid

**Comment by patw0929** [comment]
Had updated PR.
I used [`shortid`](https://www.npmjs.com/package/shortid) here, because seems that `node-uuid` depends on `crypto`.

It's a optional attribute, so developer can also set a custom `id` here.

**Comment by brunops** [comment]
@patw0929 is there any reason why a user would want to pass an internal `id`? Whatever the answer is, I think ok to make it optional, I would just remove it from the example, so the component looks simpler to use. But it's up to you :+1:

**Comment by patw0929** [comment]
@brunops: 
Like the example page in this PR: I put two `react-intl-tel-input` components here, then I can easily get specific `id`'s result from `this.state` by custom `id`. I'm not sure that is there better way to retrieve the other one's result (which has auto-generated `id`)?

I also agree that we should make this component looks simpler to use, so I will remove `id` from the example code from usage section later.

Thank you!

**Comment by brunops** [comment]
Oh, you are totally correct, it's all about accessing it from the store later on, makes sense.

**Comment by janmarek** [comment]
Hi guys,
I have a different solution which is much simpler and does not require shortid dependency. And it works very well. Every component has its own redux store instead of its own id prop.

I wanted to send a pull request, but I was not able to revert #74 pull request, because it contains also other changes (updating dependencies, rebuilds etc).

But you can verify the idea by this simple change in App.js.

```js
class IntlTelInput extends Component {
  static propTypes = {
    css: PropTypes.arrayOf(PropTypes.string),
    fieldName: PropTypes.string,
    countriesData: PropTypes.arrayOf(PropTypes.array),
    value: PropTypes.string,
    defaultValue: PropTypes.string,
    disabled: PropTypes.bool,
    allowExtensions: PropTypes.bool,
    autoFormat: PropTypes.bool,
    autoPlaceholder: PropTypes.bool,
    autoHideDialCode: PropTypes.bool,
    defaultCountry: PropTypes.string,
    geoIpLookup: PropTypes.func,
    nationalMode: PropTypes.bool,
    numberType: PropTypes.string,
    noCountryDataHandler: PropTypes.func,
    onlyCountries: PropTypes.arrayOf(PropTypes.string),
    preferredCountries: PropTypes.arrayOf(PropTypes.string),
    utilsScript: PropTypes.string,
    onPhoneNumberChange: PropTypes.func,
    onSelectFlag: PropTypes.func,
  }

  store = createStore(reducer, {}, devtool); // <----

  render() {
    return (
      <Provider store={this.store}>
        <IntlTelInputApp
          id="HARDCODED_ID" // <----
          value={this.props.value}
          countriesData={this.props.countriesData}
          defaultValue={this.props.defaultValue}
          disabled={this.props.disabled}
          onPhoneNumberChange={this.props.onPhoneNumberChange}
          onSelectFlag={this.props.onSelectFlag}
          css={this.props.css}
          fieldName={this.props.fieldName}
          allowExtensions={this.props.allowExtensions}
          autoFormat={this.props.autoFormat}
          autoPlaceholder={this.props.autoPlaceholder}
          autoHideDialCode={this.props.autoHideDialCode}
          defaultCountry={this.props.defaultCountry}
          geoIpLookup={this.props.geoIpLookup}
          nationalMode={this.props.nationalMode}
          numberType={this.props.numberType}
          noCountryDataHandler={this.props.noCountryDataHandler}
          onlyCountries={this.props.onlyCountries}
          preferredCountries={this.props.preferredCountries}
          utilsScript={this.props.utilsScript}
        />
      </Provider>
    );
  }
}
```

**Comment by patw0929** [comment]
@janmarek:
I'm quickly trying to use the idea you said, but seems that not working for me. 
Maybe I made some mistake here... so could you help to make a PR in your spare time?
Thank you. :smiley:

**Comment by janmarek** [comment]
Maybe we can help each other. Can you prepare a branch without your fix for me? :)

**Comment by patw0929** [comment]
Sure! I'll prepare it tomorrow. :)
(Have to sleep now)

**Comment by janmarek** [comment]
Thank you

**Comment by patw0929** [comment]
@janmarek: 
I have pushed a branch here [revert-dispatch-id](https://github.com/patw0929/react-intl-tel-input/tree/revert-dispatch-id). Thanks for your help!

**Pull request merged**